applications/common/backend/src/test/java/pl/ds/kyanite/common/components/models/table/TableColspansCalculationTest.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(SlingContextExtension.class)
+public class TableColspansCalculationTest {
+
+  private final SlingContext context = new SlingContext(ResourceResolverType.RESOURCERESOLVER_MOCK);
+
+  @BeforeEach
+  public void init() {
+    context.addModelsForClasses(TableCellComponent.class, TableHeadCellComponent.class, TableComponent.class);
+  }
+
+  private static Stream<Arguments> inputColumnsAndExpectedComputedColspans() {
+    return Stream.of(
+            Arguments.of(
+                    columns(4, 4, 4, null),
+                    colspans(4, 4, 4, 1)),
+            Arguments.of(
+                    columns(4, null, 4, null),
+                    colspans(4, 2, 4, 2)),
+            Arguments.of(
+                    columns(null, null, 4, null),
+                    colspans(3, 3, 4, 2)),
+            Arguments.of(
+                    columns(null, null, 5, null),
+                    colspans(3, 2, 5, 2)),
+            Arguments.of(
+                    columns(null, null, 6, null),
+                    colspans(2, 2, 6, 2)),
+            Arguments.of(
+                    columns(null, null, 7, null),
+                    colspans(2, 2, 7, 1)),
+            Arguments.of(
+                    columns(null, null, 12, null),
+                    colspans(1, 1, 12, 1)),
+            Arguments.of(
+                    columns(null, null, null, null),
+                    colspans(3, 3, 3, 3)),
+            Arguments.of(
+                    columns(3, null, null, null, 2),
+                    colspans(3, 3, 2, 2, 2)),
+            Arguments.of(
+                    columns(null, 12),
+                    colspans(1, 12)),
+            Arguments.of(
+                    columns(null, 15, null, null),
+                    colspans(1, 15, 1, 1)),
+            Arguments.of(
+                    columns(null, 11, null, null),
+                    colspans(1, 11, 1, 1)),
+            Arguments.of(
+                    columns(1, 2, 3, 4),
+                    colspans(1, 2, 3, 4)),
+            Arguments.of(
+                    columns(20),
+                    colspans(20))
+    );
+  }
+
+  private static Stream<List<Integer>> inputColumns() {
+    return inputColumnsAndExpectedComputedColspans()
+            .map(args -> args.get()[0])
+            .map(arg -> (List<Integer>) arg);
+  }
+
+  private static List<Integer> columns(Integer... columns) {
+    return Arrays.asList(columns);
+  }
+
+  private static List<Integer> colspans(int... colspans) {
+    return IntStream.of(colspans).boxed().toList();
+  }
```

Comment:
value of `columns` is nullable (can be specified by Author or not) - but the computed colspans should never be null